### PR TITLE
LDL factorization improvement

### PR DESCRIPTION
**Pull request by sshin23**
This PR improves LDLFactorization interface.

We also add a test for Float128 problems.

### CODE REVIEW

File: lib/MadNLPTests/src/Instances/hs15.jl [hunk 1]
```diff
@@ -1,9 +1,9 @@
-struct HS15Model <: NLPModels.AbstractNLPModel{Float64,Vector{Float64}}
-    meta::NLPModels.NLPModelMeta{Float64, Vector{Float64}}
+struct HS15Model{T} <: NLPModels.AbstractNLPModel{T,Vector{T}}
+    meta::NLPModels.NLPModelMeta{T, Vector{T}}
     counters::NLPModels.Counters
 end
 
-function HS15Model(; x0=zeros(2), y0=zeros(2))
+function HS15Model(T = Float64; x0=zeros(T,2), y0=zeros(T,2))
```
Comment (frapac) [inline lib/MadNLPTests/src/Instances/hs15.jl:6]: I would maybe pass `T` directly in the optional arguments after the semicolon `;`?

File: src/LinearSolvers/cholmod.jl [hunk 2]
```diff
@@ -5,7 +5,7 @@ end
 mutable struct CHOLMODSolver{T} <: AbstractLinearSolver{T}
     inner::CHOLMOD.Factor{Float64}
     tril::SparseMatrixCSC{T,Int32}
-    full::SparseMatrixCSC{Float64,Int32}
+    full::SparseMatrixCSC{Float64,Int}
```
Comment (frapac) [inline src/LinearSolvers/cholmod.jl:8]: Just ouf of curiosity, why do we have to use `Int` there?

**Review by frapac — APPROVED** [review]
I am impressed by how few changes are required to make `QuadMath` works within MadNLP. Good job! 

I just have one comment, but the PR looks good to merge.

File: src/LinearSolvers/cholmod.jl (hunk 2 above)
Comment (sshin23) [reply src/LinearSolvers/cholmod.jl:8]: For `VERSION <= v"1.9"`, cholmod throws an error if the argument is `full::SparseMatrixCSC{Float64,Int32}
`

Comment (frapac) [reply src/LinearSolvers/cholmod.jl:8]: ah, good catch

**Pull request merged**